### PR TITLE
[NFC][Opaque pointer][LLVM] Remove code in ifndef INTEL_SYCL_OPAQUEPOINTER_READY

### PR DESCRIPTION
**Pull request by jsji**
- Remove code guarded in ifndef INTEL_SYCL_OPAQUEPOINTER_READY

This should be NFC, as INTEL_SYCL_OPAQUEPOINTER_READY is always 1 since #10888 .

### CODE REVIEW

**Comment by jsji** [comment]
@aelovikov-intel Looks like we need to manually clean up some temp file in 'amdgpu-3' runner.

> fatal: update_ref failed for ref 'HEAD': cannot lock ref 'HEAD': Unable to create '/__w/repo_cache/intel/llvm/.git/HEAD.lock': File exists.

**Comment by aelovikov-intel** [comment]
> @aelovikov-intel Looks like we need to manually clean up some temp file in 'amdgpu-3' runner.
> 
> > fatal: update_ref failed for ref 'HEAD': cannot lock ref 'HEAD': Unable to create '/__w/repo_cache/intel/llvm/.git/HEAD.lock': File exists.

Done, I think. I've removed `repo_cache/intel/llvm/` and restarted the task. `cached_checkout` action recreates the cache when it's missing.

**Comment by jsji** [comment]
Failures in AMD/HIP is CI issue .

**Comment by aelovikov-intel** [comment]
> Failures in AMD/HIP is CI issue .

There isn't enough information to merge this without full testing passing.

**Comment by aelovikov-intel** [comment]
> Sure. Trying to re-run AMD/HIP dozen times without success, trying again.

I've just initiated reboot of amd-02 runner.

**Comment by aelovikov-intel** [comment]
Also, if `INTEL_SYCL_OPAQUEPOINTER_READY` is universally true, then it should be made clear in the description to highlight the "NFC" nature of this change.

Why aren't we changing any `CMakeLists.txt` files here?

**Comment by jsji** [comment]
> Also, if `INTEL_SYCL_OPAQUEPOINTER_READY` is universally true, then it should be made clear in the description to highlight the "NFC" nature of this change.

Sure, added.

> Why aren't we changing any `CMakeLists.txt` files here?

Good question, I leave them intentionally to next PR once all these PRs (https://github.com/intel/llvm/pull/11063, https://github.com/intel/llvm/pull/11062, https://github.com/intel/llvm/pull/11059) are merged -- just in case that we remove the definition and we fall back to some old code path unintentionally.

**Comment by jsji** [comment]
Ping @intel/dpcpp-tools-reviewers . Thanks.

File: llvm/include/llvm/IR/DerivedTypes.h [hunk 1]
```diff
@@ -676,16 +673,10 @@ class PointerType : public Type {
   /// given address space. This is only useful during the opaque pointer
   /// transition.
   /// TODO: remove after opaque pointer transition is complete.
-  [[deprecated("Use PointerType::get() with LLVMContext argument instead")]]
-  static PointerType *getWithSamePointeeType(PointerType *PT,
-                                             unsigned AddressSpace) {
-#ifdef INTEL_SYCL_OPAQUEPOINTER_READY
+  [[deprecated("Use PointerType::get() with LLVMContext argument "
+               "instead")]] static PointerType *
+  getWithSamePointeeType(PointerType *PT, unsigned AddressSpace) {
```
Comment (asudarsa) [inline llvm/include/llvm/IR/DerivedTypes.h:678]: Is this function ready to be removed?

Thanks

File: llvm/lib/IR/Core.cpp [hunk 2]
```diff
@@ -851,13 +851,7 @@ LLVMTypeRef LLVMPointerType(LLVMTypeRef ElementType, unsigned AddressSpace) {
...
 }
 
-LLVMBool LLVMPointerTypeIsOpaque(LLVMTypeRef Ty) {
-#ifdef INTEL_SYCL_OPAQUEPOINTER_READY
-  return true;
-#else // INTEL_SYCL_OPAQUEPOINTER_READY
-  return unwrap(Ty)->isOpaquePointerTy();
-#endif // INTEL_SYCL_OPAQUEPOINTER_READY
-}
+LLVMBool LLVMPointerTypeIsOpaque(LLVMTypeRef Ty) { return true; }
```
Comment (asudarsa) [inline llvm/lib/IR/Core.cpp:854]: I think we can remove this function and its uses? But that needs to happen in llorg though.

Thanks

**Review by asudarsa — APPROVED** [review]
Changes look good on the whole.

Thanks for putting this up.

File: llvm/lib/IR/Core.cpp (hunk 2 above)
Comment (jsji) [reply llvm/lib/IR/Core.cpp:854]: Yes, we will sync with llorg on this. We will start to remove usage of it in our own code first, after removing code guarded in INTEL_SYCL_OPAQUEPOINTER_READY. After that, we will consider removing it if community also remove it.

**Comment by jsji** [comment]
@intel/llvm-gatekeepers Can we get this merged? Thanks!

**Pull request merged**